src/imitation/algorithms/preference_comparisons.py

```diff
@@ -735,14 +966,13 @@ def forward(
                 reward model and the target probabilities in `preferences`. Metrics
                 are include accuracy.
         """
-        probs = th.empty(len(fragment_pairs), dtype=th.float32)
-        for i, fragment in enumerate(fragment_pairs):
-            frag1, frag2 = fragment
-            trans1 = rollout.flatten_trajectories([frag1])
-            trans2 = rollout.flatten_trajectories([frag2])
-            rews1 = self._rewards(model, trans1)
-            rews2 = self._rewards(model, trans2)
-            probs[i] = self._probability(rews1, rews2)
+        preference_predictor = PreferencePredictor(
```

Comment:
`PreferencePredictor` is created inside `CrossEntropyRewardLoss` (once for each call to `forward()`) but is created externally and passed through to `ActiveSelectionFragmenter`. Is there a reason to have a different convention here? Either seem fine to me but I'd expect it to be simpler to understand the code if we stick to one.